src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/TypeSpecFileService.cs

```diff
@@ -0,0 +1,118 @@
+using Azure.Tools.GeneratorAgent.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Tools.GeneratorAgent
+{
+    /// <summary>
+    /// Service responsible for retrieving TypeSpec files from various sources (local or GitHub).
+    /// </summary>
+    internal class TypeSpecFileService : IDisposable
+    {
+        private readonly AppSettings AppSettings;
+        private readonly ILogger<TypeSpecFileService> Logger;
+        private readonly ILoggerFactory LoggerFactory;
+        private readonly ValidationContext ValidationContext;
+        private GitHubFilesRetrieveService? GitHubService;
+
+        public TypeSpecFileService(
+            AppSettings appSettings,
+            ILogger<TypeSpecFileService> logger,
+            ILoggerFactory loggerFactory,
+            ValidationContext validationContext)
+        {
+            ArgumentNullException.ThrowIfNull(appSettings);
+            ArgumentNullException.ThrowIfNull(logger);
+            ArgumentNullException.ThrowIfNull(loggerFactory);
+            ArgumentNullException.ThrowIfNull(validationContext);
+
+            AppSettings = appSettings;
+            Logger = logger;
+            LoggerFactory = loggerFactory;
+            ValidationContext = validationContext;
+        }
+
+        /// <summary>
+        /// Gets TypeSpec files from either local directory or GitHub repository.
+        /// Returns a dictionary where key is filename and value is file content.
+        /// </summary>
+        public async Task<Result<Dictionary<string, string>>> GetTypeSpecFilesAsync(
+            CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrWhiteSpace(ValidationContext.ValidatedCommitId))
+            {
+                // Local case
+                return await GetLocalTypeSpecFilesAsync(ValidationContext.ValidatedTypeSpecDir, cancellationToken);
+            }
+            else
+            {
+                // GitHub case
+                return await GetGitHubTypeSpecFilesAsync(cancellationToken);
+            }
+        }
+
+        private async Task<Result<Dictionary<string, string>>> GetLocalTypeSpecFilesAsync(
+            string typeSpecDir, 
+            CancellationToken cancellationToken)
+        {
+            Logger.LogInformation("Reading TypeSpec files from local directory: {TypeSpecDir}", typeSpecDir);
+
+            try
+            {
+                Dictionary<string, string> typeSpecFiles = new Dictionary<string, string>();
+                
+                string[] allFiles = Directory.GetFiles(typeSpecDir, "*.tsp", SearchOption.AllDirectories);
```

Comment:
If I'm following the flow correctly, it seems like the local files are only pulling .tsp and the GitHub download is including TypeSpec related files too? Or am I misunderstanding?